hoomd/GSDDumpWriter.cc

```diff
@@ -63,63 +64,211 @@ GSDDumpWriter::GSDDumpWriter(std::shared_ptr<SystemDefinition> sysdef,
         throw std::invalid_argument("Invalid GSD file mode: " + mode);
         }
     m_log_writer = pybind11::none();
+
+#ifdef ENABLE_MPI
+    if (m_sysdef->isDomainDecomposed())
+        {
+        m_gather_tag_order = GatherTagOrder(m_exec_conf->getMPICommunicator());
+        }
+#endif
+
+    m_particle_dynamic.reset();
+    m_particle_dynamic[gsd_flag::position] = true;
+    m_particle_dynamic[gsd_flag::orientation] = true;
```

Comment:
According to the migration guide, these are not dynamic by default any more.

Reply:
I tried to be clear in the migration guide:
> `hoomd.write.GSD` no longer enforces 'property' as an always dynamic quantity

I did not change the default (see `gsd.py` line 178) and the corresponding docstring.

The migration guide covers the case where a user provides a value for `dynamic` that doesn't include `'property'`.

For example, in v3:
```
>>> write_gsd = hoomd.write.GSD(..., dynamic = ['momentum'])
>>> print(write_gsd.dynamic)
['property', 'momentum']
```

In v4:
>>> write_gsd = hoomd.write.GSD(..., dynamic = ['momentum'])
>>> print(write_gsd.dynamic)
['particles/velocity', 'particles/angmom', 'particles/image']
```

If you have any suggestions on how to make this more clear in the migration guide, let me know.